server/api/crud/files.py

```diff
@@ -74,3 +87,49 @@ def _get_filestat(
             "modified": stat.modified,
             "mimetype": ctype,
         }
+
+    def _delete_files(
+        self,
+        schema: str,
+        path: str,
+        user: str,
+        auth_info: mlrun.common.schemas.AuthInfo,
+        secrets: dict = None,
+        project: str = "",
+    ):
+        path = self._resolve_obj_path(schema, path, user)
+
+        logger.debug("Got delete files request", path=path)
+
+        secrets = secrets or {}
+        secrets.update(server.api.api.utils.get_secrets(auth_info))
+
+        # TODO: handle delete some files
+        obj = store_manager.object(url=path, secrets=secrets, project=project)
+        obj.delete()
+
+    @staticmethod
+    def _resolve_obj_path(schema: str, path: str, user: str):
+        path = server.api.api.utils.get_obj_path(schema, path, user=user)
+        if not path:
+            server.api.api.utils.log_and_raise(
+                HTTPStatus.NOT_FOUND.value,
+                path=path,
+                err="illegal path prefix or schema",
+            )
+        return path
+
+    @staticmethod
+    def _verify_and_get_project_secrets(project):
+        # If running on Docker or locally, we cannot retrieve project secrets, so skip.
+        if not server.api.utils.singletons.k8s.get_k8s_helper(
+            silent=True
+        ).is_running_inside_kubernetes_cluster():
+            return {}
+
+        secrets_data = server.api.crud.Secrets().list_project_secrets(
+            project,
+            mlrun.common.schemas.SecretProviderName.kubernetes,
+            allow_secrets_from_k8s=True,
+        )
+        return secrets_data.secrets or {}
```

Comment:
that's code dup, find it a better home
possibly under
server.api.utils.[project_secrets.py]
and have it optional with the opa